design/ironic-interaction/ironic-background-docs.md

```diff
@@ -0,0 +1,346 @@
+node is already in the "available" state, the same requst can be used
+with a target of "manage", and then the target of "inspect" can be utilized
+to step through the state machine.
+See: https://docs.openstack.org/ironic/latest/contributor/states.html
+
+After inspection, it is advisable to return nodes to an "available" state.
+This can be performed simillarlly via the target "provide".
+
+### How to deploy?
+
+Starting with the bare metal node in the "available" provision_state:
+
+1. First assert configuration to API to indicate the the image written
+   to disk. This is performed as a HTTP PATCH request to the
+   /v1/nodes/node-name endpoint.
+
+       {
+       {“op”: “replace”, “path”: “/instance_info”, “value”: {
+           “image_source”: “http://url-to-image”,
+           “image_os_hash_algo”: “sha256”,
+           “image_os_hash_value”: “abcdefghi…”}},
+       {“op”: “replace”, “path”: “/instance_uuid”, “value”: “anyuuidvalue”}},
+       }
+
+   **NOTE:** Instead of defining the "image_os_hash_*" values, a MD5 based
+   image checksum can be set.
+
+   This configuration does two things. First sets the image and checksum
+   to be utilized for image verification, and sets an "instance_uuid" value
+   which acts as a signal to any client that the node has been claimed by
+   an API cient. The instancce_uuid can be set to any value, and is
+   ultimately not required.
+
+2. Request ironic to perform a "validate" operation on the information
+   it is presently configured with. The expected response is a HTTP 200
+   return code, with a message body that consists of a list of "driver"
+   interfaces and any errors if applicable.
+
+    GET /v1/nodes/node-name/validate
+
+   Reply: 
+
+       {
+       "boot": true,
+       ..
+       "deploy": "configuration error message if applicable"
+       }
+
+   The particular interfaces that would be important to pay attention to are
+   ‘boot’, ‘deploy’, ‘power’, ‘management’.
+
+   More information can be found at: https://developer.openstack.org/api-ref/baremetal/?expanded=validate-node-detail
+
+3. Craft a configuration drive file
+
+   Configuration drives are files that contain a small ISO9660 filesystem
+   which contains configuration metadata and user defined "user-data".
+
+   1) Create a folder called “TEMPDIR”
+   2) In the case of ignition based configuration, that file would be
+      placed renamed "user_data" and placed in TMPDIR/openstack/latest/
+      folder.
+   3) Metadata for networking configuration setup using "cloud-init" or
+      cloud-init similar application would also be written to the
+      TEMPDIR/openstack/latest as well. This is out of scope, but is well
+      documented in the OpenStack community.
+   4) TEMPDIR is converted to an iso9660 image using a label of “config-2”.
+   5) The resulting ISO9660 image file is then compressed using the gzip
+      algorithm.
+   6) The resulting gzip image file is then encoded in base64 for storage
```

Comment:
I do not think TheJulia can bear it anymore, but I think gzip files are commonly called "archives" as opposed to images (the latter tend to resemble a snapshot of a block device).

Reply:
I lost my mind about 30 comments ago. :)  I left out the word "compressed"